lib/membrane_rtc_engine/endpoints/rtsp_endpoint.ex

```diff
@@ -38,19 +57,77 @@ if Enum.all?(
       state = %{
         rtc_engine: opts.rtc_engine,
         source_uri: opts.source_uri,
-        track: nil
+        stream_id: nil,
+        track_data: nil,
+        video: nil,
+        ssrc: nil
       }
 
-      structure = []
+      structure = [
+        child(:udp_source, %Membrane.UDP.Source{
+          local_port_no: @rtp_port,
+          recv_buffer_size: 500_000
+        })
+        |> via_in(Pad.ref(:rtp_input, make_ref()))
+        |> child(:rtp, Membrane.RTP.SessionBin)
+      ]
 
       {[spec: structure], state}
     end
 
     @impl true
-    def handle_pad_added(Pad.ref(:output, {_track_id, _variant}) = _pad, _ctx, state) do
-      %Track{encoding: encoding} = track = state.track
+    def handle_spec_started([:udp_source, :rtp], _ctx, state) do
+      connection_manager_spec = [
+        %{
+          id: "ConnectionManager",
+          start:
+            {ConnectionManager, :start_link,
+             [
+               [
+                 stream_uri: state.source_uri,
+                 port: @rtp_port,
+                 endpoint: self()
+               ]
+             ]},
+          restart: :transient
+        }
+      ]
+
+      # XXX we should probably add to an existing supervision tree instead
+      Supervisor.start_link(connection_manager_spec,
```

Comment:
Good idea but maybe let's start with linking connection manager to the endpoint and assume that if signalling crashes we shut down everything and hope that someone will restart the whole endpoint.

Users get info when some endpoint crashesh so they can restart it.